app/src/main/java/com/hedvig/app/feature/insurance/ui/InsuranceAdapter.kt

```diff
@@ -191,34 +190,39 @@ class InsuranceAdapter(
                             TypeOfContract.SE_APARTMENT_RENT,
                             TypeOfContract.NO_HOME_CONTENT_OWN,
                             TypeOfContract.NO_HOME_CONTENT_RENT,
-                            TypeOfContract.NO_TRAVEL,
-                            TypeOfContract.NO_TRAVEL_YOUTH,
-                            TypeOfContract.DK_HOME_CONTENT -> {
-                                adapter.submitList(
-                                    listOf(
-                                        ContractModel.ContractType(contract.typeOfContract),
-                                        ContractModel.NoOfCoInsured(contract.currentAgreement.numberCoInsured)
-                                    )
-                                )
-                            }
+                            TypeOfContract.DK_HOME_CONTENT,
                             TypeOfContract.NO_HOME_CONTENT_YOUTH_OWN,
                             TypeOfContract.NO_HOME_CONTENT_YOUTH_RENT,
                             TypeOfContract.SE_APARTMENT_STUDENT_BRF,
                             TypeOfContract.SE_APARTMENT_STUDENT_RENT -> {
                                 adapter.submitList(
                                     listOf(
-                                        ContractModel.ContractType(contract.typeOfContract),
-                                        ContractModel.Student(contract.typeOfContract),
+                                        ContractModel.Address(
+                                            currentAgreement = if (isHomeInsurance(contract)) {
```

Comment:
Have we not validated that the contract is a home insurance contract before we add this item to the list?

An alternative approach could be to use `listOfNotNull()` and `if (isHomeInsurance(contract) { ContractModel.Address() } else { null }` to exclude the item entirely in that case.